From 8c62aaacc73250411b9e4a3756c0ccbee96812e3 Mon Sep 17 00:00:00 2001
From: Lily Zhang <r58066@freescale.com>
Date: Tue, 13 Mar 2012 18:47:50 +0800
Subject: ENGR00177310-2 mx6 clock: change _clk_clko_round_rate

Change _clk_clko_round_rate and ensure the clock should
be less than the input rate.

Signed-off-by: Lily Zhang <r58066@freescale.com>
---
 arch/arm/mach-mx6/clock.c |    3 +++
 1 file changed, 3 insertions(+)

diff --git a/arch/arm/mach-mx6/clock.c b/arch/arm/mach-mx6/clock.c
index 06516b8..a6a1715 100644
--- a/arch/arm/mach-mx6/clock.c
+++ b/arch/arm/mach-mx6/clock.c
@@ -4821,6 +4821,9 @@ static unsigned long _clk_clko_round_rate(struct clk *clk,
 	 */
 	if (div == 0)
 		div++;
+	else if (parent_rate % rate)
+		div++;
+
 	if (div > 8)
 		div = 8;
 	return parent_rate / div;
-- 
1.7.9.5

